pipelines/toast_satellite_sim.py

```diff
@@ -356,6 +420,60 @@ def main():
             pars[ "kfilter" ] = "F"
         pars[ "fsample" ] = args.samplerate
 
+        # Simulate input sky
+
+        dist_rings = tm.DistRings(comm.comm_world,
+                            nside = args.nside,
+                            nnz = 3)
+
+        pysm_sky_config = {
+            'synchrotron': "s1",
+            'dust': "d8",
+            'freefree': "f1",
+            'cmb': "c1",
+            'ame': "a1",
+        }
+        band_center = 25
+        bandwidth = 6.3
+
+        bandpasses = {
+                "1a": (np.linspace(band_center-bandwidth/2, band_center+bandwidth/2, 10), np.ones(10)),
+                "1b": (np.linspace(band_center-bandwidth/2, band_center+bandwidth/2, 10), np.ones(10)),
+        }
+        pysm_sky = tm.PySMSky(local_pixels=dist_rings.local_pixels, nside=args.nside,
+                       pysm_sky_config=pysm_sky_config, bandpasses=bandpasses)
+        local_maps = dict()  # FIXME use Cache instead
+        pysm_sky.exec(local_maps, out="sky")
+
+        lmax = 512
+
+        beam_deg = {
+                "1a": 14/60,
+                "1b": 14/60
+        }
+
+        full_map_rank0 = {}
+        for ch_tag, fwhm_deg in beam_deg.items():
+            # LibSharp also supports transforming multiple channels together each with own beam
+            smooth = tm.LibSharpSmooth(comm.comm_world, signal_map="sky_"+ch_tag,
+                                       lmax=lmax, grid=dist_rings.libsharp_grid,
+                                       fwhm_deg=fwhm_deg, beam=None)
+            smooth.exec(local_maps)
+
+            n_components = 3
+
+            full_map_rank0[ch_tag] = assemble_map_on_rank0(comm, local_maps[ch_tag], n_components, npix)
+
+        #full_map_rank0 dict contains on rank 0 the smoothed PySM maps
```

Comment:
@tskisner I have started working on satellite sim, I am assembling the map on rank 0. My idea was to use then `scan_signal` like in the ground sim